lib/temporal/worker.rb

```diff
@@ -0,0 +1,72 @@
+require 'async'
+require 'temporal/bridge'
+require 'temporal/data_converter'
+require 'temporal/failure_converter'
+require 'temporal/payload_converter'
+require 'temporal/runtime'
+require 'temporal/worker/activity'
+require 'temporal/worker/thread_pool_executor'
+
+module Temporal
+  class Worker
+    # TODO: Add worker interceptors
+    def initialize(
+      connection,
+      namespace,
+      task_queue,
+      payload_converter: Temporal::PayloadConverter::DEFAULT,
+      payload_codecs: [],
+      failure_converter: Temporal::FailureConverter::DEFAULT,
+      activity_executor: nil,
+      max_concurrent_activities: 100
+    )
+      @running = false
+      @mutex = Mutex.new
+      @runtime = Temporal::Runtime.instance
+      activity_executor ||= ThreadPoolExecutor.new(max_concurrent_activities)
```

Comment:
Just want to :+1: this as the default

Reply:
You'll get starvation here, change it to:

```suggestion
      activity_executor ||= ThreadPoolExecutor.new(max_concurrent_activities + max_concurrent_local_activities)
```

Those are 2 separate knobs.
As an optimization you'd want an autoscaling thread pool but that can be handled later (not sure what this implementation does).

Reply:
I looked, it allocates threads up front.

Reply:
Yep, will fix this #76